query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -403,31 +401,20 @@ final case class LabelCardinalityExec(queryContext: QueryContext,
     source.acquireSharedLock(dataset, shard, querySession)
     val rvs = source match {
       case memstore: TimeSeriesStore =>
-        val shardKeyCols = memstore.schemas(dataset).get.part.options.shardKeyColumns.map(_.utf8)
-
-        // TODO: require presence of shard key in column filters
-        val matches = memstore.partKeysWithFilters(dataset, shard, filters, fetchFirstLastSampleTimes = false,
-          endMs, startMs, limit = 1).toSeq
-        if (matches.nonEmpty) {
+        // first get the list of label names, then fetch each label value and add to CPC sketch
+        val labelNames = memstore.labelNames(dataset, shard, filters, endMs, startMs)
+        if (labelNames.nonEmpty) {
           val sketchMap = scala.collection.mutable.Map[String, CpcSketch]()
-          val firstMatchLVs = matches.head
-          val shardKeyLVs = firstMatchLVs.filterKeys(shardKeyCols.contains)
-          shardKeyLVs.foreach { case (label, value) =>
-            sketchMap.getOrElseUpdate(label.toString, new CpcSketch(logK)).update(value.toString)
-          }
-
-          val nonShardKeyLVs = firstMatchLVs.filterKeys(k => !shardKeyCols.contains(k))
-          nonShardKeyLVs.foreach { case (label, _) =>
-            val labelStr = label.toString
+          labelNames.foreach { case label =>
             // GOTCHA: This approach will not catch cardinality of labels which are disabled for faceting
             // since their value lengths are > 1000. We expect the gateway to reject (or shorten) that data early on.
             memstore.singleLabelValueWithFilters(dataset, shard, filters, label.toString,
               endMs, startMs, querySession, 1000000).foreach { labelValue =>
-              sketchMap.getOrElseUpdate(labelStr, new CpcSketch(logK)).update(labelValue.toString)
+              sketchMap.getOrElseUpdate(label, new CpcSketch(logK)).update(labelValue.toString)
             }
           }
 
-          val rv = IteratorBackedRangeVector(CustomRangeVectorKey(shardKeyLVs.toMap),
+          val rv = IteratorBackedRangeVector(CustomRangeVectorKey.empty,
```

Comment:
@amolnayak311 this is okay? Double-checking since it looks like this was intentional (logic exists elsewhere to handle empty/nonempty keys).